driver.go

```diff
@@ -257,10 +258,114 @@ func findNeighbors(tx neo4j.Transaction, ids []uint64) ([]neo4jNode, []neo4jNode
+	// check if value type exists in config
+	for k, v := range input {
+		if k != cfg.Map(ls.NodeValueTerm) {
+			continue
+		}
+		vt := cfg.Map(cfg.PropertyTypeMappings[k])
+		if vt != "" {
+			va := ls.GetValueAccessor(vt)
+			switch val := v.(type) {
+			case bool, neo4j.Time, neo4j.LocalTime, neo4j.LocalDateTime, neo4j.Date:
+				if va != nil {
+					form, err := va.FormatNativeValue(val, val, node)
+					if err != nil {
+						panic(fmt.Errorf("cannot format native value for %v, %w", v, err))
+					}
+					return form
+				} else {
+					return val
+				}
+			case float64:
+				if va != nil {
+					form, err := va.FormatNativeValue(val, val, node)
+					if err != nil {
+						panic(fmt.Errorf("cannot format native value for %v, %w", v, err))
+					}
+					return form
+				} else {
+					f := strconv.FormatFloat(v.(float64), 'f', -1, 64)
+					return f
+				}
+			default:
+				return v
+			}
+		} else {
+			switch val := v.(type) {
+			case neo4j.LocalDateTime:
+				x := val.Time()
+				tm := types.DateTime{
+					Month:        int(x.Month()),
+					Year:         x.Year(),
+					Day:          x.Day(),
+					Seconds:      int64(x.Second()),
+					Minute:       int64(x.Minute()),
+					Milliseconds: int64(x.Second() / 1000),
+					Hour:         int64(x.Hour()),
+					Nanoseconds:  int64(x.Nanosecond()),
+					Location:     x.Location(),
+				}
+				return tm
+				// if err := ls.SetNodeValue(node, tm); err != nil {
+				// 	panic(fmt.Errorf("Error setting node value for %v, %w", node, err))
+				// }
+			case neo4j.LocalTime:
+				x := val.Time()
+				tm := types.TimeOfDay{
+					Seconds:      int64(x.Second()),
+					Milliseconds: int64(x.Second() / 1000),
+					Hour:         int64(x.Hour()),
+					Minute:       int64(x.Minute()),
+					Nanoseconds:  int64(x.Nanosecond()),
+					Location:     x.Location(),
+				}
+				return tm
+				// if err := ls.SetNodeValue(node, tm); err != nil {
+				// 	panic(fmt.Errorf("Error setting node value for %v, %w", node, err))
+				// }
+			case neo4j.Date:
+				x := val.Time()
+				tm := types.Date{
+					Month:    int(x.Month()),
+					Day:      x.Day(),
+					Year:     x.Year(),
+					Location: x.Location(),
+				}
+				return tm
+			// if err := ls.SetNodeValue(node, tm); err != nil {
+			// 	panic(fmt.Errorf("Error setting node value for %v, %w", node, err))
+			// }
+
+			default:
+				val, err := ls.GetNodeValue(node)
+				if err != nil {
+					panic(fmt.Errorf("Cannot get node value for %v %w", node, err))
+				}
+				return val
+				// return v
+			}
+		}
+	}
+	return nil
+}
+
+func MakePropertiesLoad(node graph.Node, input map[string]interface{}, cfg Config) map[string]*ls.PropertyValue {
 	ret := make(map[string]*ls.PropertyValue)
 	for k, v := range input {
+		// check if there is a type for property in config, otherwise convert to string and store it
+		if k == ls.NodeValueTerm {
+			continue
+		}
```

Comment:
Check if the expanded term has a type mapping here. If so, use the value accessor for that type.